pkg/controller/daemon/daemon_controller.go

```diff
@@ -1032,7 +1032,7 @@ func (dsc *DaemonSetsController) syncNodes(ds *apps.DaemonSet, podsToDelete, nod
 
 				podTemplate := &template
 
-				if utilfeature.DefaultFeatureGate.Enabled(features.ScheduleDaemonSetPods) {
+				if utilfeature.DefaultFeatureGate.Enabled(features.TaintNodesByCondition) {
```

Comment:
Should not make it dependent on `TaintNodesByCondition`, it breaks ScheduleDaemonSetPods features :)

Reply:
ok, I will just leave it as is.